External/FEXCore/Source/Interface/IR/Passes/DeadCodeElimination.cpp

```diff
@@ -38,12 +38,22 @@ bool DeadCodeElimination::Run(IREmitter *IREmit) {
     // We grab these nodes this way so we can iterate easily
     auto CodeBegin = CurrentIR.at(BlockIROp->Begin);
     auto CodeLast = CurrentIR.at(BlockIROp->Last);
+    auto CodeEnd = CurrentIR.at(BlockIROp->Last);
 
     while (1) {
       auto CodeOp = CodeLast();
       OrderedNode *CodeNode = CodeOp->GetNode(ListBegin);
       auto IROp = CodeNode->Op(DataBegin);
 
+      // A BFE that extract all bits is a nop
+      if (IROp->Op == OP_BFE) {
```

Comment:
Would make more sense to move this to the Constant Propagation pass.

Reply:
Will make a new pass for these cases - also `AND %v, %v`, `OR %v, %v`, `XOR %v, %v`, `Add/Sub %v, 0`, etc

Reply:
I think it makes sense to put these in ConstProp; We already optimise away OP_LSHL shift by zero there.

What is best for performance, a single pass which does both, or two passes?

Reply:
Done in #338